frontend/src/metabase/core/components/TabList/TabList.tsx

```diff
@@ -23,20 +27,72 @@ const TabList = forwardRef(function TabGroup<T>(
 ) {
   const idPrefix = useUniqueId();
   const outerContext = useContext(TabContext);
+
+  const [scrollPosition, setScrollPosition] = useState(0);
+  const [showScrollLeft, setShowScrollLeft] = useState(false);
+  const [showScrollRight, setShowScrollRight] = useState(false);
+
+  const tabListContentRef = useRef(null);
+
   const innerContext = useMemo(() => {
     return { value, idPrefix, onChange };
   }, [value, idPrefix, onChange]);
+
   const activeContext = outerContext.isDefault ? innerContext : outerContext;
 
+  const scroll = (direction: string) => {
+    if (tabListContentRef.current) {
+      const container = tabListContentRef.current as HTMLDivElement;
+
+      const scrollDistance =
+        (container.offsetWidth - 32) * (direction === "left" ? -1 : 1);
+      container.scroll(container.scrollLeft + scrollDistance, 0);
```

Comment:
Should we use `scrollBy` to make this a little more concise?

```suggestion
      container.scrollBy(scrollDistance, 0);
```

This helps micro-optimize a bit too by calling `.scrollLeft` 1 less time as `.scrollLeft` causes a reflow. But I believed it's negligible in the real world though.